lib/Froxlor/Validate/Validate.php

```diff
@@ -62,32 +62,78 @@ public static function validate($str, $fieldname, $pattern = '', $lng = '', $emp
-	 * @param string $lng
-	 *        	index for error-message (if $return_bool is false)
-	 * @param bool $allow_localhost
-	 *        	whether to allow 127.0.0.1
-	 * @param bool $allow_priv
-	 *        	whether to allow private network addresses
-	 * @param bool $allow_cidr
-	 *        	whether to allow CIDR values e.g. 10.10.10.10/16
-	 *        	
-	 * @return string|bool ip address on success, false on failure
-	 */
-	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $throw_exception = false)
+    /**
+     * Converts CIDR to a netmask address
+     *
+     * @thx to https://stackoverflow.com/a/5711080/3020926
+     * @param string $cidr
+     *
+     * @return string
+     */
+    public static function cidr2NetmaskAddr ($cidr) {
+
+        $ta = substr ($cidr, strpos ($cidr, '/') + 1) * 1;
+        $netmask = str_split (str_pad (str_pad ('', $ta, '1'), 32, '0'), 8);
+
+        foreach ($netmask as &$element) {
+            $element = bindec ($element);
+        }
+
+        return implode ('.', $netmask);
+    }
+
+    /**
+     * Checks if an $address (IP) is IPv6
+     *
+     * @param $address
+     *
+     * @return bool
+     */
+    public static function is_ipv6($address) {
+        return filter_var($address, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6);
+    }
+
+    /**
+     * Checks whether it is a valid ip
+     *
+     * @param string $ip
+     *            ip-address to check
+     * @param bool   $return_bool
+     *            whether to return bool or call \Froxlor\UI\Response::standard_error()
+     * @param string $lng
+     *            index for error-message (if $return_bool is false)
+     * @param bool   $allow_localhost
+     *            whether to allow 127.0.0.1
+     * @param bool   $allow_priv
+     *            whether to allow private network addresses
+     * @param bool   $allow_cidr
+     *            whether to allow CIDR values e.g. 10.10.10.10/16
+     * @param bool   $cidr_as_netmask
+     *            whether to format CIDR nodation to netmask notation
+     *
+     * @param bool   $throw_exception
+     *            whether to throw an exception on failure
+     *
+     * @return string|bool ip address on success, false on failure
+     */
+	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $cidr_as_netmask = false, $throw_exception = false)
 	{
 		$cidr = "";
 		if ($allow_cidr) {
 			$org_ip = $ip;
 			$ip_cidr = explode("/", $ip);
-			if (count($ip_cidr) == 2) {
+			if (count($ip_cidr) === 2) {
+			    if(strlen($ip_cidr[1]) <= 2 && in_array((int)$ip_cidr[1], array_values(range(1, 32)), TRUE) === false) {
+                    \Froxlor\UI\Response::standard_error($lng, $ip, $throw_exception);
+                }
+                if ($cidr_as_netmask && self::is_ipv6($ip_cidr[0])) {
```

Comment:
now there's `is_ipv6()` for only one occurence, might as well do this and remove `is_ipv6()` completely 
```diff
- if ($cidr_as_netmask && self::is_ipv6($ip_cidr[0])) {
+ if ($cidr_as_netmask && filter_var($ip_cidr[0], FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)) {
```

Reply:
Maybe I can add some unit tests for that. Would that help? I think its best if we have this capsuled in case its needed sometime down the road.

Reply:
It's used in various places, could be helpful to have that capsuled but then use it everywhere where `filter_var($xyz, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)` is used (counting 18 occurences using grep)

Reply:
We can change that. But lets tackle this in a new PR.

Reply:
agreed